src/room.cpp

```diff
@@ -78,6 +78,29 @@ void Room::HandleJoinRequest(const RakNet::Packet* packet) {
     SendJoinSuccess(member);
 }
 
+void Room::HandleChatPacket(const RakNet::Packet* packet) {
+    RakNet::BitStream in_stream(packet->data, packet->length, false);
+
+    in_stream.IgnoreBytes(sizeof(RakNet::MessageID));
+    RakNet::RakString message;
+    in_stream.Read(message);
+    auto CompareNetworkAddress = [&](const Member member) -> bool {
+        return member.network_address == packet->systemAddress;
+    };
+    const auto sending_member = std::find_if(members.begin(), members.end(), CompareNetworkAddress);
+
+    if (sending_member == members.end())    // Sender is not a joined member
+        return;
```

Comment:
we should `assert();` here instead imo. Receiving a packet from a non connected person should not be possible